### PR TITLE
Hover for POI icons

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/web/issues/459

File: map/src/map/layers/PoiLayer.js [hunk 1]
```diff
@@ -70,6 +73,32 @@ export async function createPoiLayer({ ctx, poiList = [], globalPoiIconCache, ty
...
+        createHoverMarker({
+            marker,
+            setSelectedId: ctx.setSelectedPoiId,
+            mainStyle: true,
+            text: marker.options['web_poi_name'],
+            latlng: marker._latlng,
+            iconSize: [DEFAULT_ICON_SIZE, DEFAULT_ICON_SIZE],
+            map,
+            ctx,
+            pointerStyle: `${styles.hoverPointer}`,
```
Comment (RZR-UA) [inline map/src/map/layers/PoiLayer.js:86]: please avoid using ` for simple strings

File: map/src/map/layers/PoiLayer.js [hunk 2]
```diff
@@ -70,6 +73,32 @@ export async function createPoiLayer({ ctx, poiList = [], globalPoiIconCache, ty
...
+
+    simpleMarkersArr.getLayers().forEach((marker) => {
+        createHoverMarker({
+            marker,
+            setSelectedId: ctx.setSelectedPoiId,
+            text: marker.options['web_poi_name'],
+            latlng: marker._latlng,
+            map,
+            ctx,
+            pointerStyle: `${styles.hoverPointer}`,
```
Comment (RZR-UA) [inline map/src/map/layers/PoiLayer.js:98]: please avoid using ` for simple strings

File: map/src/map/layers/SearchLayer.js [hunk 3]
```diff
@@ -38,6 +40,10 @@ export function findFeatureGroupById(map, id) {
     return foundGroup;
 }
 
+export function getObjIdSearch(obj) {
+    return obj.properties['web_poi_id'] ?? obj.geometry.coordinates[1] + obj.geometry.coordinates[0];
```
Comment (RZR-UA) [inline map/src/map/layers/SearchLayer.js:44]: strange consturction. is return value a string or a sum of double values?

File: map/src/map/layers/SearchLayer.js [hunk 4]
```diff
@@ -219,6 +232,32 @@ export default function SearchLayer() {
...
+            createHoverMarker({
+                marker,
+                setSelectedId: ctx.setSelectedPoiId,
+                mainStyle: true,
+                text: marker.options['web_poi_name'] ?? marker.options['web_name'],
+                latlng: marker._latlng,
+                iconSize: [DEFAULT_ICON_SIZE, DEFAULT_ICON_SIZE],
+                map,
+                ctx,
+                pointerStyle: `${styles.hoverPointer}`,
```
Comment (RZR-UA) [inline map/src/map/layers/SearchLayer.js:245]: `

File: map/src/map/layers/SearchLayer.js [hunk 5]
```diff
@@ -219,6 +232,32 @@ export default function SearchLayer() {
...
+
+        simpleMarkersArr.getLayers().forEach((marker) => {
+            createHoverMarker({
+                marker,
+                setSelectedId: ctx.setSelectedPoiId,
+                text: marker.options['web_poi_name'] ?? marker.options['web_name'],
+                latlng: marker._latlng,
+                map,
+                ctx,
+                pointerStyle: `${styles.hoverPointer}`,
```
Comment (RZR-UA) [inline map/src/map/layers/SearchLayer.js:257]: `

File: map/src/map/util/Clusterizer.js [hunk 6]
```diff
@@ -110,3 +111,110 @@ export function createSecondaryMarker(obj) {
...
+    map,
+    ctx,
+    pointerStyle = `${styles.wikiIconHover} ${styles.wikiIconLarge}`,
+}) {
+    let tooltipRef = ctx.searchTooltipRef;
+    let pointerRef = ctx.searchPointerRef;
+
+    map.on('zoomend', () => {
+        removeTooltip(map, tooltipRef);
+    });
```
Comment (RZR-UA) [inline map/src/map/util/Clusterizer.js:131]: where is the corresponding map.off ?

File: map/src/map/util/TrackLayerProvider.js [hunk 7]
```diff
@@ -182,7 +183,7 @@ function getPointGeoProfile(point, points) {
     }
 }
 
-function parseWpt(points, layers, ctx = null, data) {
+function parseWpt(points, layers, ctx = null, data, map) {
```
Comment (RZR-UA) [inline map/src/map/util/TrackLayerProvider.js:186]: please use parseWpt({ ... })

File: map/src/menu/search/search/SearchResultItem.jsx [hunk 8]
```diff
@@ -83,7 +104,9 @@ export default function SearchResultItem({ item, setSearchValue }) {
                 <div>
                     <MenuItem
                         id={id}
-                        className={styles.searchItem}
+                        onMouseEnter={() => handleMouseEnter()}
+                        onMouseLeave={() => handleMouseLeave()}
```
Comment (RZR-UA) [inline map/src/menu/search/search/SearchResultItem.jsx:108]: () => is redundant here

**Review by RZR-UA — APPROVED** [review]

**Pull request merged**